pulp_rpm/tests/functional/constants.py

```diff
@@ -198,6 +201,101 @@
 }
 """The metadata for one RPM package."""
 
+
+RPM_COMPLEX_PACKAGE_DATA = {
+    "arch": "x86_64",
+    "artifact": None,
+    "changelogs": [
+        [
+            "Lucille Bluth <lucille@bluthcompany.com> - 1.1.1-1",
+            1617192000,
+            "- It's a banana, Michael. How much could it cost, $10?",
+        ],
+        [
+            "Job Bluth <job@alliance-of-magicians.com> - 2.2.2-2",
+            1619352000,
+            "- I've made a huge mistake",
+        ],
+        [
+            "George Bluth <george@federalprison.gov> - 3.3.3-3",
+            1623672000,
+            "- There’s always money in the banana stand",
+        ],
+    ],
+    "checksum_type": "sha256",
+    "conflicts": [["foxnetwork", "GT", "0", "5555", None, False]],
+    "description": "Complex package",
+    "enhances": [["(bananas or magic)", None, None, None, None, False]],
+    "epoch": "1",
+    "files": [
+        [None, "/etc/complex/", "pkg.cfg"],
+        [None, "/usr/bin/", "complex_a"],
+        ["dir", "/usr/share/doc/", "complex-package"],
+        [None, "/usr/share/doc/complex-package/", "README"],
+        ["dir", "/var/lib/", "complex"],
+        ["ghost", "/var/log/", "complex.log"],
+    ],
+    "is_modular": False,
+    "location_base": "",
+    "location_href": "complex-package-2.3.4-5.el8.x86_64.rpm",
+    "md5": None,
+    "name": "complex-package",
+    "obsoletes": [
+        ["bluemangroup", "LT", "0", "32.1", "0", False],
+        ["cornballer", "LT", "0", "444", None, False],
+    ],
+    "pkg_id": "ff872c221377fe2ac26dcf4c0b5c7376e46764ed8ddbb6215863510a11cd7e0f",
+    "provides": [
+        ["complex-package", "EQ", "1", "2.3.4", "5.el8", False],
+        ["complex-package(x86-64)", "EQ", "1", "2.3.4", "5.el8", False],
+        ["laughter", "EQ", "0", "33", None, False],
+        ["narration(ronhoward)", None, None, None, None, False],
+    ],
+    "recommends": [
+        ["((hiding and attic) if light-treason)", None, None, None, None, False],
+        ["GeneParmesan(PI)", None, None, None, None, False],
+        ["yacht", "GT", "9", "11.0", "0", False],
+    ],
+    "release": "5.el8",
+    "requires": [
+        ["/usr/sbin/useradd", None, None, None, None, True],
+        ["arson", "GE", "0", "1.0.0", "1", False],
+        ["fur", "LE", "0", "2", None, False],
+        ["staircar", "LE", "0", "99.1", "3", False],
+    ],
+    "rpm_buildhost": "fedora.redhat.com",
+    "rpm_group": "Development/Tools",
+    "rpm_header_end": 8393,
+    "rpm_header_start": 4504,
+    "rpm_license": "MPLv2",
+    "rpm_packager": "Michael Bluth",
+    "rpm_sourcerpm": "complex-package-2.3.4-5.el8.src.rpm",
+    "rpm_vendor": "Bluth Company",
+    "sha1": None,
+    "sha224": None,
+    "sha256": None,
```

Comment:
This is interesting - is it a bug that this value isn't showing up?  It's an on-demand content unit but the sha256 checksum was provided.

Reply:
It looks like the serializer just grabs these checksums from the artifact. We could update it to be intelligent and also check the remote artifact.